classes/class-wc-rest-connect-base-controller.php

```diff
@@ -63,14 +71,44 @@ public function get_internal( $request ) {
 		if ( ! defined( 'DONOTCACHEPAGE' ) ) {
 			define( 'DONOTCACHEPAGE', true ); // Play nice with WP-Super-Cache
 		}
-		return $this->get( $request );
+		try {
+			return $this->get( $request );
+		} catch ( WCC_Api_Exception $error ) {
+			return $error->response;
+		}
 	}
 
 	public function post_internal( $request ) {
 		if ( ! defined( 'DONOTCACHEPAGE' ) ) {
 			define( 'DONOTCACHEPAGE', true ); // Play nice with WP-Super-Cache
 		}
-		return $this->post( $request );
+		try {
+			return $this->post( $request );
+		} catch ( WC_Connect_Api_Exception $ex ) {
+			return $ex->response;
+		}
+	}
+
+	protected function api_request( $method /*, ...$args */ ) {
+		$args = array_slice( func_get_args(), 1 );
+		$response = call_user_func_array( array( $this->api_client, $method ), $args );
+
+		if ( is_wp_error( $response ) ) {
+			$error = new WP_Error(
+				$response->get_error_code(),
+				$response->get_error_message(),
+				array( 'message' => $response->get_error_message() )
+			);
+			$this->logger->debug( $error, get_class( $this ) );
+			throw new WC_Connect_Api_Exception( $error );
+		}
+
+		if ( isset( $response->async_token ) || isset( $response->async_status ) ) {
+			// Not an error per-se, but we want to interrupt further processing of the response at this point
```

Comment:
There's no more processing *in this function*, but there is in its caller. The endpoint calls `$this->api_request()`, and after that it processes the response (which would be the list of purchased labels). So throwing an exception here ensures that the rest of the endpoint code won't be run. That's why I said the this is "using exceptions as flow control", which is not very good practice. It allows the endpoint to work without any changes though.